sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -408,6 +416,35 @@ public String getId() {
         return queryItemsInternal(new SqlQuerySpec(query), new CosmosQueryRequestOptions(), classType);
     }
 
+    /**
+     * Initializes the container by warming up the caches and connections.
+     * @return Mono of Void
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Mono<Void> initializeContainerAsync() {
+        if(this.getDatabase().getClient().getDocClientWrapper().getConnectionPolicy().getConnectionMode().equals(ConnectionMode.GATEWAY)) {
+            logger.info("Client is created with gateway mode, initialization of container is not needed, no action triggered");
+            return Mono.empty();
+        }
+
+        Mono<List<FeedRange>> feedRangesMono = this.getDatabase().getClient().getDocClientWrapper().getFeedRanges(this.getLink());
+        AtomicReference<Mono<List<FeedResponse<ObjectNode>>>> sequentialList = new AtomicReference<>();
+        List<Flux<FeedResponse<ObjectNode>>> fluxList = new ArrayList<>();
+        return feedRangesMono.flatMap(feedRanges -> {
+            for (FeedRange feedRange : feedRanges) {
+                String dummyQuery = String.format("SELECT * from c where c.id = '%s'", UUID.randomUUID().toString());
+                CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+                options.setConsistencyLevel(ConsistencyLevel.STRONG);
+                options.setFeedRange(feedRange);
+                CosmosPagedFlux<ObjectNode> cosmosPagedFlux = this.queryItems(dummyQuery, options,
```

Comment:
This particular operation consumes RUs; translation: the customer is charged $$ for something that he might not be aware of. We need appropriate documentation to capture this. Also from the implementation point of view please be sure this is the cheapest way to initialize these connections.

Reply:
Done as per offline conversation